AppSettingsManagerAPI/AppSettingsManagerApi.Infrastructure/MySql/MySqlSettingRepository.cs

```diff
@@ -8,116 +8,118 @@ namespace AppSettingsManagerApi.Infrastructure.MySql;
 
 public class MySqlSettingRepository : ISettingRepository
 {
-    // Creates SettingsContext object
     private readonly SettingsContext _settingsContext;
-    private readonly IBidirectionalConverter<Model.Setting, Setting> _settingsConverter;
+    private readonly IBidirectionalConverter<Model.Setting, Setting> _settingConverter;
+    private readonly IBidirectionalConverter<
+        Model.SettingGroup,
+        SettingGroup
+    > _settingGroupConverter;
 
     public MySqlSettingRepository(
         SettingsContext settingsContext,
-        IBidirectionalConverter<Model.Setting, Setting> settingsConverter
+        IBidirectionalConverter<Model.Setting, Setting> settingConverter,
+        IBidirectionalConverter<Model.SettingGroup, SettingGroup> settingGroupConverter
     )
     {
         // Injects SettingsContext configured in ServiceConfiguration.cs into _settingsContext object
         _settingsContext = settingsContext;
-        // Injects biverter that we set up into _settingsConverter object
-        _settingsConverter = settingsConverter;
+        _settingConverter = settingConverter;
+        _settingGroupConverter = settingGroupConverter;
     }
 
-    public async Task<Model.Setting> GetSetting(string settingId, int version)
+    public async Task<Model.SettingGroup> GetSettingGroup(string settingGroupId)
     {
-        // Call .Single() because there should only be one entry with this id/version
-        // and .Single() will return a single object rather than a list
-        var setting = await _settingsContext.Settings.SingleAsync(
-            s => s.SettingGroupId == settingId && s.Version == version
-        );
+        var settingGroup = await GetSettingGroupFromContext(settingGroupId);
 
-        return _settingsConverter.Convert(setting);
+        return _settingGroupConverter.Convert(settingGroup);
     }
 
-    public async Task<IEnumerable<Model.Setting>> GetAllSettingVersions(string settingId)
+    public async Task<IEnumerable<Model.SettingGroup>> GetSettingGroupsByUser(string userId)
     {
-        var settings = await GetAllUnconvertedSettingVersions(settingId);
-
-        // This could also be written as settings.Select(s => _settingsConverter.Convert(s))
+        var settingGroups = await _settingsContext.Permissions
```

Comment:
These could realistically just be a return statement but probably easier to debug having it broken up a bit. Shouldn't change performance